atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java

```diff
@@ -191,7 +191,14 @@ public RaftContext(
     lastVotedFor = meta.loadVote();
 
     // Construct the core log, reader, writer, and compactor.
-    raftLog = storage.openLog(meta, threadContext);
+    // very much unclear if we want to redirect uncaught exceptions here and step down all the time
+    raftLog =
```

Comment:
Not a fan of this. Needs to be cleaned up a little, how the thread factory is provided.